src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/sampler/HTTPHCAbstractImpl.java

```diff
@@ -107,8 +110,13 @@
     static {
         if(!StringUtils.isEmpty(JMeterUtils.getProperty("httpclient.timeout"))) { //$NON-NLS-1$
             log.warn("You're using property 'httpclient.timeout' that will soon be deprecated for HttpClient3.1, you should either set "
-                    + "timeout in HTTP Request GUI, HTTP Request Defaults or set http.socket.timeout in httpclient.parameters");
+                    + "timeout in HTTP Request GUI, HTTP Request Defaults or set http.socket.timeout in hc.parameters");
+        }
+
+        if(!StringUtils.isEmpty(JMeterUtils.getProperty("hc.parameters.file"))) { //$NON-NLS-1$
+            HttpClientDefaultParameters.load(JMeterUtils.getProperty("hc.parameters.file"), HTTPCLIENT_PARAMS);
```

Comment:
It looks like the file will be loaded only once. I suggest the file should be loaded at the start of the test.

Reply:
@vlsi this is an existing feature flow where this file is loaded once along with jmeter properties as per design. Not sure if I unserstand that correct to change this.